### PR TITLE
OJ-2320: added alarms for the dead Letter event queue and the audit e…

### PR DESCRIPTION
**Pull request by liamStockwellGDS**
…vent queue

NINO | Platform | Add alarms for AuditQueue and Dead Letter Queue

### What changed

added alarms to template file.

### Why did it change

To allow for alarms 

### Issue tracking

- [OJ-2320](https://govukverify.atlassian.net/browse/OJ-2320)

### Environment variables or secrets

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-2320]: https://govukverify.atlassian.net/browse/OJ-2320?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -1837,6 +1837,61 @@ Resources:
...
+      TreatMissingData: missing
+      Dimensions:
+        - Name: RuleName
+          Value: !Select [ 1, !Split [ "|", !Ref TxMaAuditEventRule ] ]
+        - Name: EventBusName
+          Value: !Ref CheckHmrcEventBus
+
+  AuditEventDeadLetterQueueAlarm:
+    Condition: "DeployAlarms"
+    "Type": "AWS::CloudWatch::Alarm"
```
Comment (surajkumar) [inline infrastructure/template.yaml:1867]: Can you remove the `"` please around these keys

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -1837,6 +1837,61 @@ Resources:
...
+    "Properties":
+      "AlarmName": !Sub "${AWS::StackName}-${Environment}-AuditEventDeadLetterQueueAlarm:"
+      "AlarmDescription": !Sub "${AuditEventResponseReceivedRule} has a request get through to the dead letter queue request in the last hour. ${SupportManualURL}"
+      "ActionsEnabled": true
+      OKActions:
+        - !ImportValue platform-alarm-warning-alert-topic
+      AlarmActions:
+        - !ImportValue platform-alarm-warning-alert-topic
+      "InsufficientDataActions": [ ]
+      "MetricName": "NumberOfMessagesReceived"
```
Comment (surajkumar) [inline infrastructure/template.yaml:1877]: Maybe we should use the `DeadLetterInvocations` metric - let me know what you think

File: infrastructure/template.yaml [hunk 3]
```diff
@@ -1837,6 +1837,61 @@ Resources:
       Threshold: 0
       TreatMissingData: missing
 
+  AuditEventExceededAgeLimitAlarm:
```
Comment (surajkumar) [inline infrastructure/template.yaml:1840]: Should this be on the SQS queue not EventBridge and EventBridge to have it's own set of alarms?

File: infrastructure/template.yaml [hunk 4]
```diff
@@ -35,7 +35,7 @@ Parameters:
   DeployAlarmsInDevEnvironment:
     Description: "Set to the string value `true` to deploy alarms in a DEV environment"
     Type: String
-    Default: false
+    Default: "true"
```
Comment (surajkumar) [inline infrastructure/template.yaml:38]: Please revert this

**Review by surajkumar — CHANGES_REQUESTED** [review]
Let's review this as maybe this isn't the correct place for the alarms to be defined. Additionally, please provide evidence of test.
Also, speak with Steve once merged as these need to be documented in the runbooks

File: infrastructure/template.yaml (hunk 3 above)
Comment (liamStockwellGDS) [reply infrastructure/template.yaml:1840]: Caitlin raised the same thing, there is a branch in ipv-cri-common-infrastructure

File: infrastructure/template.yaml (hunk 2 above)
Comment (liamStockwellGDS) [reply infrastructure/template.yaml:1877]: if we're now going to attach directly SQS Queue, `DeadLetterInvocations` could be useful, bav used `ApproximateNumberOfMessagesVisible` for theirs

**Pull request closed without merging**